src/ucp/wireup/select.c

```diff
@@ -566,18 +566,33 @@ static UCS_F_NOINLINE ucs_status_t ucp_wireup_select_transport(
     return UCS_OK;
 }
 
-static inline double ucp_wireup_tl_iface_latency(ucp_context_h context,
-                                                 const uct_iface_attr_t *iface_attr,
-                                                 const ucp_address_iface_attr_t *remote_iface_attr)
+static double
+ucp_wireup_fp8_pack_unpack_latency(const ucp_context_h context,
+                                   const ucs_linear_func_t *latency)
+{
+    double lat_nsec    = ucp_tl_iface_latency(context, latency) *
+                         UCS_NSEC_PER_SEC;
+    uint8_t packed_lat = UCS_FP8_PACK(LATENCY, lat_nsec);
+
+    return UCS_FP8_UNPACK(LATENCY, packed_lat) / UCS_NSEC_PER_SEC;
+}
+
+static inline double
+ucp_wireup_tl_iface_latency(ucp_context_h context,
+                            const uct_iface_attr_t *iface_attr,
+                            const ucp_address_iface_attr_t *remote_iface_attr)
 {
+    double unpacked_lat;
+
     if (remote_iface_attr->addr_version == UCP_OBJECT_VERSION_V1) {
-        /* Address v1 contains just latency overhead */
-        return ucs_max(iface_attr->latency.c, remote_iface_attr->lat_ovh) +
-               (iface_attr->latency.m * context->config.est_num_eps);
+        unpacked_lat = ucp_tl_iface_latency(context, &iface_attr->latency);
     } else {
-        return ucs_max(remote_iface_attr->lat_ovh,
-                       ucp_tl_iface_latency(context, &iface_attr->latency));
+        /* FP8 is a lossy compression method, so in order to create a symmetric
+         * calculation we pack/unpack the local latency as well */
+        unpacked_lat = ucp_wireup_fp8_pack_unpack_latency(context,
+                                                          &iface_attr->latency);
     }
+    return ucs_max(remote_iface_attr->lat_ovh, unpacked_lat);
```

Comment:
`remote_iface_attr->lat_ovh` can contain just `iface_attr->latency.c` portion

Reply:
but this change increases probability for ep reconfig for address v1, while the PR fixes an issue with address v2 (off by default)

Reply:
i think it also should fix ep reconfiguration with address v1 (and not increase the error probability)
since in v1 we take max of remote and num_eps-adjusted-local, we better also adjust before packing the value we provide to remove peer.

Reply:
@yosefe, afair we had similar problem with BW, which we had to fix here:
https://github.com/openucx/ucx/blob/master/src/ucp/wireup/address.c#L818

Reply:
the fix was passing BW after considering PPN: https://github.com/openucx/ucx/blob/master/src/ucp/wireup/address.c#L655
Therefore IMO we should do the same with latency packing